src/storage/async_page_fetcher.cpp

```diff
@@ -47,6 +64,21 @@ namespace cublog
     m_callback (logreader.get_page (), err);
   }
 
+  void data_page_fetch_task::execute (context_type &context)
+  {
+    // TODO: wait for replication
```

Comment:
@cristiarg merged the changes that were required for this todo. Please add the code that waits for the replication.